File: controllers/upload_controller.js
const express = require('express');
const router = express.Router();
const fs = require('fs')

router.get('/list', (req, res) => {
    const files = fs.readdirSync("./uploads")
    return res.status(200).send(files)
})

router.get('/rm/:path', async (req, res) => {
    path = 'uploads/' + req.params.path
    if (fs.existsSync(path)) {
        fs.unlinkSync(path)
        const files = fs.readdirSync("./uploads")
        return res.status(200).send(files)
    } else {
        const files = fs.readdirSync("./uploads")
        return res.status(404).send(files)
    }
})

router.post('/', (req, res) => {
    if (req.files === null) {
        return res.status(400).json({ msg: 'No file was uploaded' })
    }

    const file = req.files.file
    file.mv(`${__dirname}/../uploads/${file.name}`, err => {
        if (err) {
            console.error(err)
            return res.status(500).send(err)
        }

        res.json({
            fileName: file.name,
            filePath: `/uploads/${file.name}`
        })
    })
})

module.exports = router

File: controllers/upload_controller.py
from werkzeug import secure_filename
from flask import Blueprint, jsonify, request
import os

upload_controller = Blueprint('upload_controller', __name__)

# List all Files
@upload_controller.route('/list', methods=['GET'])
def upload_list():
    res = os.listdir('./uploads')
    valid_ext = {'.jpg', '.png', '.mov'}
    res = [item for item in res if item[-4:] in valid_ext]
    return jsonify(res)

# Remove Uploaded File
@upload_controller.route('/rm/<path:path>', methods=['GET'])
def remove_file(path):
    status_code = 404
    full_path = os.path.join('uploads', path)
    if os.path.exists(full_path):
        os.remove(full_path)
        status_code = 200

    res = os.listdir('./uploads')
    valid_ext = {'.jpg', '.png', '.mov'}
    res = [item for item in res if item[-4:] in valid_ext]
    return jsonify(res), status_code

# File Uploader
@upload_controller.route('', methods=['POST'])
def upload_file():
    f = request.files['file']
    file_save_path = os.path.join('uploads', secure_filename(f.filename))
    f.save(file_save_path)
    return 'file uploaded successfully'
